pkg/yttlibrary/overlay/map_item_match_annotation.go

```diff
@@ -50,7 +50,15 @@ func (a MapItemMatchAnnotation) Index(leftMap *yamlmeta.Map) (int, bool, error)
 		count = 1
 	}
 
-	return idx, found, a.expects.Check(count)
+	err := a.expects.Check(count)
+	if err != nil {
+		if found {
+			err = fmt.Errorf("Found item: %s", err)
```

Comment:
check errors shouldnt be augmented here as only Check() inside knows what expected (found item vs not found).